src/mon/OSDMonitor.cc

```diff
@@ -1162,16 +1231,16 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
   int r = pending_inc.propagate_snaps_to_tiers(g_ceph_context, osdmap);
   assert(r == 0);
 
-  if (mapping && mapping_job) {
+  if (mapping_job) {
     if (!mapping_job->is_done()) {
       dout(1) << __func__ << " skipping prime_pg_temp; mapping job "
 	      << mapping_job.get() << " did not complete, "
 	      << mapping_job->shards << " left" << dendl;
       mapping_job->abort();
-    } else if (mapping->get_epoch() == osdmap.get_epoch()) {
+    } else if (mapping.get_epoch() == osdmap.get_epoch()) {
```

Comment:
hi @tchaikov @liewegas , the condition here should be `mapping.get_epoch() < osdmap.get_epoch()` ? 
i think we can only prime pg temp when the mapping is up to date, am i wrong ?